### PR TITLE
Feature/moxy

### PR DESCRIPTION
**Pull request by georgious070**

### CODE REVIEW

File: app/src/main/java/com/example/android/marinaforecastinc/ui/detail/DetailForecastPresenter.java [hunk 1]
```diff
@@ -8,9 +8,13 @@
...
     private DetailForecastView detailForecastView;
 
-    public DetailForecastPresenter(BaseView baseView) {
-        super(baseView);
-        detailForecastView = (DetailForecastView) baseView;
+    public DetailForecastPresenter(){
+    }
+
+    @Override
+    public BaseView getViewState() {
```
Comment (georgious070) [inline app/src/main/java/com/example/android/marinaforecastinc/ui/detail/DetailForecastPresenter.java]: dont look at it

File: app/src/main/java/com/example/android/marinaforecastinc/ui/detail/DetailForecastActivity.java [hunk 2]
```diff
@@ -5,15 +5,25 @@
...
 import com.example.android.marinaforecastinc.ui.base.BaseActivity;
 
 public class DetailForecastActivity extends BaseActivity implements DetailForecastView {
 
+    @InjectPresenter
+    public DetailForecastPresenter presenter;
+
     private TextView textCityDetail;
-    private DetailForecastPresenter presenter;
     private static final String KEY_CITY_NAME = "cityName";
```
Comment (MarinaShaposhnikova) [inline app/src/main/java/com/example/android/marinaforecastinc/ui/detail/DetailForecastActivity.java]: move constant field to the top

File: app/src/main/java/com/example/android/marinaforecastinc/ui/detail/DetailForecastActivity.java [hunk 3]
```diff
@@ -5,15 +5,25 @@
...
 
+    @ProvidePresenter
+    DetailForecastPresenter providePresenter(){
+        presenter = new DetailForecastPresenter(getIntent().getStringExtra(KEY_CITY_NAME));
+        return presenter;
```
Comment (MarinaShaposhnikova) [inline app/src/main/java/com/example/android/marinaforecastinc/ui/detail/DetailForecastActivity.java]: It's a mistake. Your presenter has already initted. Don't init it again.
Use : 
return new DetailForecastPresenter(getIntent().getStringExtra(KEY_CITY_NAME));;

File: app/src/main/java/com/example/android/marinaforecastinc/ui/detail/DetailForecastPresenter.java [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
 
+@InjectViewState
+public class DetailForecastPresenter extends BasePresenter<DetailForecastView> {
 
-public class DetailForecastPresenter extends BasePresenter {
-
-    private DetailForecastView detailForecastView;
-
-    public DetailForecastPresenter(){
+    String cityName;
```
Comment (MarinaShaposhnikova) [inline app/src/main/java/com/example/android/marinaforecastinc/ui/detail/DetailForecastPresenter.java]: Find the mistake here, please

File: app/src/main/java/com/example/android/marinaforecastinc/ui/detail/DetailForecastPresenter.java [hunk 5]
```diff
@@ -1,17 +1,20 @@
...
+    public DetailForecastPresenter(String cityName){
+        this.cityName = cityName;
     }
 
-    public void viewLoaded(String cityName) {
-        detailForecastView.showCityName(cityName);
+    @Override
+    public void attachView(DetailForecastView view) {
+        super.attachView(view);
+        view.showCityName(cityName);
```
Comment (MarinaShaposhnikova) [inline app/src/main/java/com/example/android/marinaforecastinc/ui/detail/DetailForecastPresenter.java]: NO!
Read about attachView and onFirstViewAttach methods

File: app/src/main/java/com/example/android/marinaforecastinc/ui/base/BasePresenter.java [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
+import com.arellomobile.mvp.MvpPresenter;
+import com.arellomobile.mvp.MvpView;
 
-    private BaseView baseView;
+public abstract class BasePresenter<T extends BaseView> extends MvpPresenter<T> {
 
-    public BasePresenter(BaseView baseView) {
-        this.baseView = baseView;
-    }
+    public BasePresenter() {
```
Comment (MarinaShaposhnikova) [inline app/src/main/java/com/example/android/marinaforecastinc/ui/base/BasePresenter.java]: Remove empty constructor

File: app/src/main/java/com/example/android/marinaforecastinc/ui/home/ForecastActivity.java [hunk 7]
```diff
@@ -40,4 +42,9 @@ public void showData(List<String> cityList) {
     public void onClickCity(String cityName) {
         startActivity(DetailForecastActivity.getIntent(ForecastActivity.this, cityName));
     }
+
+    @Override
+    public void showToast(String message) {
+
+    }
```
Comment (MarinaShaposhnikova) [inline app/src/main/java/com/example/android/marinaforecastinc/ui/home/ForecastActivity.java]: What is it???

**Pull request merged**